tox.ini

```diff
@@ -209,103 +145,74 @@ commands = {toxinidir}/scripts/check_packages.py
 [testenv:docs-serve]
 skipsdist = True
 skip_install = True
-deps =
-    bs4==0.0.1
-    jinja2==3.0.3
-    markdown==3.3.6
-    mkdocs==1.1
-    mkdocs-material==6.2.8
-    pymdown-extensions==9.1
-commands = pip3 install git+https://github.com/pugong/mkdocs-mermaid-plugin.git#egg=mkdocs-mermaid-plugin
-           mkdocs build --clean
-           python -c 'print("###### Starting local server. Press Control+C to stop server ######")'
-           mkdocs serve -a localhost:8080
+commands = 
+  poetry install --only docs
+  poetry run mkdocs serve -a localhost:8080
+           
 
 [testenv:flake8]
 skipsdist = True
 skip_install = True
-deps =
-    flake8==3.7.9
-    flake8-bugbear==20.1.4
-    flake8-docstrings==1.6.0
-    flake8-eradicate==0.4.0
-    flake8-isort==4.0.0
-    pydocstyle==3.0.0
-commands = flake8 aea benchmark examples packages plugins scripts tests
+commands = 
+  poetry run python ./install_packages.py "flake8.*" pydocstyle
+  poetry run flake8 aea benchmark examples packages plugins scripts tests
 
 [testenv:liccheck]
 skipsdist = True
-usedevelop = True
-deps =
-    liccheck==0.6.0
-commands = {toxinidir}/scripts/freeze_dependencies.py -o {envtmpdir}/requirements.txt
-           liccheck -s strategy.ini -r {envtmpdir}/requirements.txt -l PARANOID
+commands = 
+  poetry install --only packages,main
+  poetry run python ./install_packages.py liccheck
+  poetry run {toxinidir}/scripts/freeze_dependencies.py -o {envtmpdir}/tmp-requirements.txt
+  poetry run liccheck -s strategy.ini -r {envtmpdir}/tmp-requirements.txt -l PARANOID
 
 [testenv:mypy]
 skipsdist = True
 skip_install = True
-deps =
-    aiohttp>=3.7.4,<3.8
-    base58>=1.0.3
-    bech32==1.2.0
-    mypy==0.761
-    packaging>=20.4,<=21.5
-    web3==5.12.0
-    yoti==2.14.0
-    asn1crypto==1.4.0
-commands = mypy aea packages --disallow-untyped-defs
-           mypy benchmark examples  --check-untyped-defs
-           mypy scripts tests
+commands = 
+    poetry install --only packages,main,testing
+    poetry run python ./install_packages.py mypy click
+    poetry run mypy aea packages --disallow-untyped-defs
+    poetry run mypy benchmark examples  --check-untyped-defs
+    poetry run mypy scripts tests
 
 [testenv:pylint]
 whitelist_externals = /bin/sh
 skipsdist = True
-deps =
-    pylint==2.6.0
-    pytest==7.0.0
-    gitpython>=3.1.14
-
 commands =
-    python -m pip install --no-deps file://{toxinidir}/plugins/aea-ledger-ethereum
-    python -m pip install --no-deps file://{toxinidir}/plugins/aea-ledger-cosmos
-    python -m pip install --no-deps file://{toxinidir}/plugins/aea-ledger-fetchai
-    python -m pip install --no-deps file://{toxinidir}/plugins/aea-cli-ipfs
-    sh -c "pylint aea benchmark packages plugins/aea-ledger-fetchai/aea_ledger_fetchai plugins/aea-ledger-ethereum/aea_ledger_ethereum plugins/aea-ledger-cosmos/aea_ledger_cosmos plugins/aea-cli-ipfs/aea_cli_ipfs scripts examples/*"
+    poetry run python ./install_packages.py pylint pytest gitpython
```

Comment:
why pytest and gitpython?

Reply:
pytest gitpython installed for scripts and aea testing tools source code check